$MirOS: ports/devel/cvs/patches/patch-doc_cvs_texinfo,v 1.3 2007/06/06 13:46:44 tg Exp $
--- doc/cvs.texinfo.orig	Fri Sep 23 02:02:31 2005
+++ doc/cvs.texinfo	Sun Sep  6 11:31:05 2009
@@ -1,10 +1,12 @@
 \input texinfo  @c -*-texinfo-*-
 @comment Documentation for CVS.
 @setfilename cvs.info
+@afourpaper
+@comment $MirOS: src/gnu/usr.bin/cvs/doc/cvs.texinfo,v 1.9 2009/09/05 18:33:37 tg Exp $
 @macro copyleftnotice
 @noindent
 Copyright @copyright{} 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000,
-                       2001, 2002, 2003, 2004, 2005
+                       2001, 2002, 2003, 2004, 2005, 2006, 2009
                        Free Software Foundation, Inc.
 
 @multitable @columnfractions .12 .88
@@ -2236,7 +2238,7 @@ proxy.  @var{hostname} should be the nam
 through and @var{port} is the port number on the HTTP proxy server to connect
 via.  @var{port} defaults to 8080.
 
-@strong{NOTE: An HTTP proxy server is not the same as a @sc{cvs} write proxy
+@emph{NOTE: An HTTP proxy server is not the same as a @sc{cvs} write proxy
 server - please see @ref{Write proxies} for more on @sc{cvs} write proxies.}
 
 For example, to connect pserver via a web proxy listening on port 8000 of
@@ -2246,7 +2248,7 @@ www.myproxy.net, you would use a method 
 :pserver;proxy=www.myproxy.net;proxyport=8000:@var{pserver_connection_string}
 @end example
 
-@strong{NOTE: In the above example, @var{pserver_connection_string} is still
+@emph{NOTE: In the above example, @var{pserver_connection_string} is still
 required to connect and authenticate to the CVS server, as noted in the
 upcoming sections on password authentication, @code{gserver}, and
 @code{kserver}.  The example above only demonstrates a modification to the
@@ -2675,7 +2677,7 @@ encouraged.  Please send a mail to one o
 (@code{info-cvs@@nongnu.org} or @code{bug-cvs@@nongnu.org}) if you use the 
 @sc{cvs} PAM support.
 
-@strong{WARNING: Using PAM gives the system administrator much more 
+@emph{WARNING: Using PAM gives the system administrator much more 
 flexibility about how @sc{cvs} users are authenticated but 
 no more security than other methods.  See below for more.} 
 
@@ -3994,7 +3996,7 @@ place.  Therefore, one might delete, mov
 tag.
 
 @noindent
-@strong{WARNING: the commands in this section are
+@emph{WARNING: the commands in this section are
 dangerous; they permanently discard historical
 information and it can be difficult or impossible to
 recover from errors.  If you are a @sc{cvs}
@@ -4024,7 +4026,7 @@ of branch tags.  In that case, any non-b
 trigger warnings and will not be deleted.
 
 @noindent
-@strong{WARNING: Moving branch tags is very dangerous!  If you think
+@emph{WARNING: Moving branch tags is very dangerous!  If you think
 you need the @code{-B} option, think again and ask your @sc{cvs}
 administrator about it (if that isn't you).  There is almost certainly
 another way to accomplish what you want to accomplish.}
@@ -4055,7 +4057,7 @@ In that case, non-branch tags encountere
 name are ignored with a warning message.
 
 @noindent
-@strong{WARNING: Moving branch tags is very dangerous!  If you think you
+@emph{WARNING: Moving branch tags is very dangerous!  If you think you
 need the @code{-B} option, think again and ask your @sc{cvs}
 administrator about it (if that isn't you).  There is almost certainly
 another way to accomplish what you want to accomplish.}
@@ -4904,7 +4906,7 @@ merging the changes between them into th
 directory need not change anything.  Therefore, there
 is no conflict.
 
-@strong{WARNING: In versions of @sc{cvs} prior to 1.12.2, there was a
+@emph{WARNING: In versions of @sc{cvs} prior to 1.12.2, there was a
 major problem with using @samp{-kk} on merges.  Namely, @samp{-kk}
 overrode any default keyword expansion mode set in the archive file in
 the repository.  This could, unfortunately for some users, cause data
@@ -7182,6 +7184,11 @@ See @ref{Configuring keyword expansion} 
 @item $@splitrcskeyword{Date}$
 The date and time (UTC) the revision was checked in.
 
+@cindex Mdocdate keyword
+@item $@splitrcskeyword{Mdocdate}$
+The date (UTC) the revision was checked in, in a format suitable
+for the Berkeley mdoc macro processing.
+
 @cindex Header keyword
 @item $@splitrcskeyword{Header}$
 A standard header containing the full pathname of the
@@ -7551,12 +7558,13 @@ A list may be used. The this example:
 
 @example
         # Add a "MyBSD" keyword and restrict keyword
-        # expansion to the MyBSD, Name and Date keywords.
+        # expansion to the MyBSD, Name, Date and Mdocdate keywords.
         LocalKeyword=MyBSD=CVSHeader
-        KeywordExpand=iMyBSD,Name,Date
+        KeywordExpand=iMyBSD,Name,Date,Mdocdate
 @end example
 
-would allow $@splitrcskeyword{MyBSD}$, $@splitrcskeyword{Name}$, and
+would allow $@splitrcskeyword{MyBSD}$, $@splitrcskeyword{Name}$,
+$@splitrcskeyword{Mdocdate} and
 $@splitrcskeyword{Date}$ to be expanded.
 
 It is also possible to configure an exclusion list
@@ -7742,7 +7750,7 @@ $ cd wdiff-0.05
 $ cvs import -m "Import of FSF v. 0.05" fsf/wdiff FSF_DIST WDIFF_0_05
 @end example
 
-@strong{WARNING: If you use a release tag that already exists in one of the
+@emph{WARNING: If you use a release tag that already exists in one of the
 repository archives, files removed by an import may not be detected.}
 
 For files that have not been modified locally, the newly created
@@ -7786,7 +7794,12 @@ should do a merge like this even if @cod
 
 You can also revert local changes completely and return
 to the latest vendor release by changing the `head'
-revision back to the vendor branch on all files.  For
+revision back to the vendor branch on all files.  This
+does, however, produce weird results if you should ever
+edit this file again, for anyone looking at the output
+from the @code{log} command or CVSweb.  To fix this,
+first commit a revision of the file which equals the
+vendor branch, then use @code{admin} @samp{-b}. For
 example, if you have a checked-out copy of the sources
 in @file{~/work.d/wdiff}, and you want to revert to the
 vendor's version for all the files in that directory,
@@ -7853,6 +7866,10 @@ one vendor, you may specify the @samp{-b
 @code{cvs import}.  It takes as an argument the vendor
 branch to import to.  The default is @samp{-b 1.1.1}.
 
+Vendor branches can only be in the format 1.1.x where
+@samp{x} is an @emph{uneven} number, because branch
+tags use even numbers.
+
 For example, suppose that there are two teams, the red
 team and the blue team, that are sending you sources.
 You want to import the red team's efforts to branch
@@ -8498,7 +8515,7 @@ option as in other commands.  (Other com
 which are listed with the individual commands, may have
 different behavior from one @sc{cvs} command to the other).
 
-@strong{Note: the @samp{history} command is an exception; it supports
+@emph{Note: the @samp{history} command is an exception; it supports
 many options that conflict even with these standard options.}
 
 @table @code
@@ -8610,7 +8627,7 @@ give an error if you mistype a tag name.
 @code{annotate}, @code{checkout}, @code{export},
 @code{rdiff}, @code{rtag}, and @code{update}.
 
-@strong{WARNING:  The @code{commit} and @code{remove}
+@emph{WARNING:  The @code{commit} and @code{remove}
 commands also have a
 @samp{-f} option, but it has a different behavior for
 those commands.  See @ref{commit options}, and
@@ -8631,7 +8648,7 @@ The @samp{-k} option is available with t
 @code{checkout}, @code{diff}, @code{export}, @code{import},
 @code{rdiff}, and @code{update} commands.
 
-@strong{WARNING: Prior to CVS version 1.12.2, the @samp{-k} flag
+@emph{WARNING: Prior to CVS version 1.12.2, the @samp{-k} flag
 overrode the @samp{-kb} indication for a binary file.  This could
 sometimes corrupt binary files.  @xref{Merging and keywords}, for
 more.}
@@ -8660,7 +8677,7 @@ Do not run any tag program.  (A program 
 specified to run in the modules
 database (@pxref{modules}); this option bypasses it).
 
-@strong{Note: this is not the same as the @samp{cvs -n}
+@emph{Note: this is not the same as the @samp{cvs -n}
 program option, which you can specify to the left of a cvs command!}
 
 Available with the @code{checkout}, @code{commit}, @code{export},
@@ -8688,6 +8705,7 @@ and @code{watchers}.
 @item -r @var{tag}[:@var{date}]
 @cindex HEAD, special tag
 @cindex BASE, special tag
+@cindex BASE, special date
 Use the revision specified by the @var{tag} argument (and the @var{date}
 argument for the commands which accept it) instead of the
 default @dfn{head} revision.  As well as arbitrary tags defined
@@ -8735,12 +8753,18 @@ When a command expects a specific revisi
 the name of a branch is interpreted as the most recent
 revision on that branch.
 
+As a MirOS CVS extension, specifying @samp{BASE} as the @var{date} portion
+of the argument yields the @emph{base revision} of the branch specified by
+the @var{tag} portion of the argument, i.e. the revision on the parent
+branch the @var{tag} branch split off, or, where both branches were the
+same.  This option has not received very much testing, beware!
+
 Specifying the @samp{-q} global option along with the
 @samp{-r} command option is often useful, to suppress
 the warning messages when the @sc{rcs} file
 does not contain the specified tag.
 
-@strong{Note: this is not the same as the overall @samp{cvs -r} option,
+@emph{Note: this is not the same as the overall @samp{cvs -r} option,
 which you can specify to the left of a @sc{cvs} command!}
 
 @samp{-r @var{tag}} is available with the @code{commit} and @code{history}
@@ -9501,7 +9525,11 @@ that the argument file contains the log 
 
 At @code{commit}, a unique commitid is placed in the @sc{rcs}
 file inside the repository. All files committed at once
-get the same commitid. The commitid can be retrieved with
+get the same commitid, a string consisting only of hexadecimal
+digits (usually 16 in GNU CVS, 19 in MirOS BSD and MirPorts GNU CVS).
+FSF GNU CVS 1.11, MirOS GNU CVS 1.11, and OpenBSD OpenCVS do not
+support commitids yet.
+The commitid can be retrieved with
 the @code{log} and @code{status} command; see @ref{log},
 @ref{File status}.
 
@@ -10425,7 +10453,7 @@ all possible events to a single history 
 (@file{$CVSROOT/CVSROOT/history}) with read and write permissions for all
 users (@pxref{Creating a repository}).
 
-@strong{Note: @code{history} uses @samp{-f}, @samp{-l},
+@emph{Note: @code{history} uses @samp{-f}, @samp{-l},
 @samp{-n}, and @samp{-p} in ways that conflict with the
 normal use inside @sc{cvs} (@pxref{Common options}).}
 
@@ -10805,7 +10833,7 @@ in local time at the client. This is typ
 the @code{$TZ} environment variable, which can be set to
 govern how @code{log} displays dates.
 
-@strong{Note: @code{log} uses @samp{-R} in a way that conflicts
+@emph{Note: @code{log} uses @samp{-R} in a way that conflicts
 with the normal use inside @sc{cvs} (@pxref{Common options}).}
 
 @menu
@@ -11282,7 +11310,7 @@ Delete your working copy of the file if 
 succeeds.  If this flag is not given your files will
 remain in your working directory.
 
-@strong{WARNING:  The @code{release} command deletes
+@emph{WARNING:  The @code{release} command deletes
 all directories and files recursively.  This
 has the very serious side-effect that any directory
 that you have created inside your checked-out sources,
@@ -11730,6 +11758,7 @@ Keywords (@pxref{Keyword list}):
 @example
 $@splitrcskeyword{Author}: joe $
 $@splitrcskeyword{Date}: 1993/12/09 03:21:13 $
+$@splitrcskeyword{Mdocdate}: December 9 1993 $
 $@splitrcskeyword{CVSHeader}: files/file1,v 1.1 1993/12/09 03:21:13 joe Exp harry $
 $@splitrcskeyword{Header}: /home/files/file1,v 1.1 1993/12/09 03:21:13 joe Exp harry $
 $@splitrcskeyword{Id}: file1,v 1.1 1993/12/09 03:21:13 joe Exp harry $
@@ -11877,7 +11906,8 @@ Use head revision if tag/date not found.
 @item -j @var{tag}[:@var{date}]
 Merge in the change specified by @var{tag}, or when @var{date} is specified
 and @var{tag} is a branch tag, the version from the branch @var{tag} as it
-existed on @var{date}.  See @ref{checkout options}.
+existed on @var{date}.  See @ref{checkout options}.  Also,
+see @ref{Common options}.
 
 @item -k @var{kflag}
 Use @var{kflag} keyword expansion.  See
@@ -11907,7 +11937,7 @@ behavior}.
 @item -r @var{tag}[:@var{date}]
 Checkout the revision already tagged with @var{tag} or, when @var{date} is
 specified and @var{tag} is a branch tag, the version from the branch @var{tag}
-as it existed on @var{date}.  This .  See @ref{Common options}.
+as it existed on @var{date}.  See @ref{Common options}.
 
 @item -s
 Like -c, but include module status.  See @ref{checkout options}.
@@ -12975,7 +13005,7 @@ two versions of the files, and require t
 mechanisms outside @sc{cvs}, to insert any necessary
 changes.
 
-@strong{WARNING: do not use @code{COPY} with
+@emph{WARNING: do not use @code{COPY} with
 @sc{cvs} 1.9 or earlier - such versions of @sc{cvs} will
 copy one version of your file over the other, wiping
 out the previous contents.}
@@ -14442,6 +14472,7 @@ Unique Session ID of the @sc{cvs} proces
 random string of printable characters of at least 16
 characters length. Users should assume that it may
 someday grow to at most 256 characters in length.
+Currently, MirPorts GNU CVS uses 19 characters.
 @end table
 
 If you want to pass a value to the administrative files
@@ -14613,7 +14644,7 @@ all uses of @code{cvs import} on the rep
 @cindex KeywordExpand, in CVSROOT/config
 @item KeywordExpand=@var{value}
 Specify @samp{i} followed by a list of keywords to be expanded
-(for example, @samp{KeywordExpand=iMYCVS,Name,Date}),
+(for example, @samp{KeywordExpand=iMYCVS,Name,Date,Mdocdate}),
 or @samp{e} followed by a list of keywords not to be expanded
 (for example, @samp{KeywordExpand=eCVSHeader}).
 For more on keyword expansion, see @ref{Configuring keyword expansion}.
@@ -14747,7 +14778,7 @@ that the file should be checked with the
 @samp{stat()} function to see if it has changed (see warning below)
 before rereading.  The default value is @samp{always}.
 
-@strong{Note: the `stat' mode can cause CVS to pause for up to
+@emph{Note: the `stat' mode can cause CVS to pause for up to
 one extra second per directory committed.  This can be less IO and
 CPU intensive but is not recommended for use with large repositories}
 
@@ -15805,6 +15836,15 @@ The list of contributors here is not com
 complete list of who has contributed to this manual see
 the file @file{doc/ChangeLog} in the @sc{cvs} source
 distribution.
+
+The MirOS Project uses @sc{cvs} heavily in MirOS BSD and the
+MirPorts Framework and has enhanced it. Responsible:
+
+@display
+Thorsten Glaser <@t{tg@@mirbsd.org}>
+@end display
+
+CVS Homepage: @url{http://www.nongnu.org/cvs/}
 
 @c ---------------------------------------------------------------------
 @node BUGS
